### ISSUE TITLE
Add settings for VM Options

### ISSUE THREAD

**Issue by Xerus2000** [sghpjuikit/player#65]
Using `ManagementFactory.getPlatformMXBean(com.sun.management.HotSpotDiagnosticMXBean::class.java` you can obtain an instance of HotSpotDiagnosticMXBean. With that you can
- get `diagnosticOptions`, which is a List of all modifieable VMOptions
- call `setVMOption` with the key and a value to adjust this option at runtime

Within the settings we could then provide a "VM Options" tab within a Group called "Dev" or "Danger zone" or the like, which lists all those options and allows you to modify them. Accessing these settings for the first time should issue a warning that modifying these can make the application unstable and that you should really know what you are doing.

**Comment by Xerus2000** [comment]
Using `ManagementFactory.getPlatformMXBean(com.sun.management.HotSpotDiagnosticMXBean::class.java` you can obtain an instance of HotSpotDiagnosticMXBean. With that you can
- get `diagnosticOptions`, which is a List of all modifieable VMOptions
- call `setVMOption` with the key and a value to adjust this option at runtime

Within the settings we could then provide a "VM Options" tab within a Group called "Dev" or "Danger zone" or the like, which lists all those options and allows you to modify them. Accessing these settings for the first time should issue a warning that modifying these can make the application unstable and that you should really know what you are doing.

**Comment by sghpjuikit** [comment]
There already is a 'developer mode' (in general settings) that you can enable. Perhaps configs should provide additional possible EditMode - DEV_ONLY in addition to existing NONE, APP, USER.

What benefit is there to changing VM options in runtime?

**Comment by sghpjuikit** [comment]
@Xerus2000 Pls comment

**Comment by Xerus2000** [comment]
I think my I was mentioning this because with the new JVM you can change the Garbage collection and RAM parameters at runtime and them having an effect.

**Comment by sghpjuikit** [comment]
Why is that desired?

**Comment by Xerus2000** [comment]
See #46

**Comment by sghpjuikit** [comment]
I do not believe users need to concern themselves with gc tuning.

However, if I have time I may implement this. There would be a special window for vmoptions, displaying both editable and read-only options, requiring checking a 'yes I wanna blow things up' checkbox. This window will only be openable by executing an Action, which means it will appear in settings and app search as a command.

I may consider dropping this if it will require using internal APIs.

**Comment by sghpjuikit** [comment]
I implemented working prototype as described above.

Works, but isn't as user friendly as normal settings (means no input validation, no observability, needs manual value change) - I did what I could though
- mimicked settings UX
- refresh value after applying
- display vmOption origin

Concerns:
- I had some issue enabling this action only in developer mode so it is enabled always for now (although then there is still disabled ui and manual unlock required)
- This relies on HotSpotDiagnosticMXBean, which requires JDK. We do not care at this point, but it is not ideal. Just saying...
- We could implement the rest of the beans, but that would require some work (UI related), graphs and polling at an interval as well as providing either a unified view or at least unified place to launch these. Not worth it.
